consensus/src/rand/rand_gen/reliable_broadcast_state.rs

```diff
@@ -95,57 +95,49 @@ impl<S: Share, P: Proof<Share = S>, D: AugmentedData>
     }
 }
 
-pub struct ShareAckState<P> {
+pub struct ShareAggregateState<S, P, Storage> {
     rand_metadata: RandMetadata,
-    validators: Mutex<HashSet<Author>>,
+    rand_store: Arc<Mutex<RandStore<S, P, Storage>>>,
```

Comment:
will this Mutex cause performance issue?

Reply:
since this always requires a write, RwLock won't be better.

Reply:
I was worrying about this blocking the rand_manager main loop. Especially it seems it's not sharded by round, so there could be unbounded number of things competing for the same lock..

Maybe it won't be a big problem, we can worry about it later.

Reply:
makes sense, i was wondering having per round mutex too